ophyd/epics_compat.py

```diff
@@ -0,0 +1,236 @@
+                    args.usr = lambda orig_cb=args.usr, **kwargs: \
+                        self.queue.put((orig_cb, [], kwargs))
+                    args.usr._disp_tag = self
+
+        return epics.ca._onMonitorEvent(args)
+
+
+def setup(logger=None):
+    '''Setup ophyd for use
+
+    Must be called once per session using ophyd
+    '''
+    import epics
+    # It's important to use the same context in the callback dispatcher
+    # as the main thread, otherwise not-so-savvy users will be very
+    # confused
+    global _dispatcher
+
+    if _dispatcher is not None:
+        logger.debug('ophyd already setup')
+        return
+
+    def _cleanup():
+        '''Clean up the ophyd session'''
+        global _dispatcher
+        if _dispatcher is None:
+            return
+
+        logger.debug('Performing ophyd cleanup')
+        if _dispatcher.is_alive():
+            logger.debug('Joining the dispatcher thread')
+            _dispatcher.stop()
+            _dispatcher.join()
+
+        _dispatcher = None
+
+        logger.debug('Finalizing libca')
+        epics.ca.finalize_libca()
+
+    epics.ca.use_initial_context()
+
+    logger.debug('Installing monitor dispatcher')
+    _dispatcher = MonitorDispatcher()
+    atexit.register(_cleanup)
+    return _dispatcher
+
+
+def get_pv_form(version):
+    '''Get the PV form that should be used for pyepics
+
+    Due to a bug in certain versions of PyEpics, form='time' cannot be used
+    with some large arrays.
+
+    native: gives time.time() timestamps from this machine
+    time: gives timestamps from the PVs themselves
+
+    Returns
+    -------
+    {'native', 'time'}
+    '''
+    def _fix_git_versioning(in_str):
+        return in_str.replace('-g', '+g')
+
+    def _naive_parse_version(version):
+        try:
+            version = version.lower()
+
+            # Strip off the release-candidate version number (best-effort)
+            if 'rc' in version:
+                version = version[:version.index('rc')]
+
+            version_tuple = tuple(int(v) for v in version.split('.'))
+        except Exception:
+            return None
+
+        return version_tuple
+
+    try:
+        from pkg_resources import parse_version
+    except ImportError:
+        parse_version = _naive_parse_version
+
+    version = parse_version(_fix_git_versioning(version))
+
+    if version is None:
+        warnings.warn('Unrecognized PyEpics version; using local timestamps',
+                      ImportWarning)
+        return 'native'
+
+    elif version <= parse_version('3.2.3'):
+        warnings.warn('PyEpics versions <= 3.2.3 will use local timestamps (version: %s)' %
+                      epics.__version__,
+                      ImportWarning)
+        return 'native'
+    else:
+        return 'time'
+
+
+pv_form = get_pv_form(epics.__version__)
+
+_exposed_names = ['caput', 'caget', 'thread_class', 'get_pv', 'setup', 'PV',
```

Comment:
How is this different from `__all__`?

Reply:
I have change it to `__all__` in uncommited changes already, this is notes to my self